### PR TITLE
[NDM][NDMII-2748] add snmp worload app

### PR DESCRIPTION
**Pull request by jedupau**
What does this PR do?
---------------------
Add snmp workload app.

Which scenarios this will impact?
-------------------

Motivation
----------
Following [this comment](https://github.com/DataDog/datadog-agent/pull/26003#discussion_r1617569451)
It will be used by the snmp new e2e test in the agent to avoid docker hub rate limit

Additional Notes
----------------

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -68,6 +68,8 @@ jobs:
             image: netflow-generator
           - app: jmxfetch
             image: jmx-test-app
+          - app: snmp
+            image: snmp
```
Comment (TCheruy) [inline .github/workflows/docker-publish.yml:72]: `image: snmpsim` or `image: snmp-simulator` ?

**Review by TCheruy — COMMENTED** [review]
~~Could you please add why this is needed in the PR descr?~~
I am blind 😭

**Review by TCheruy — APPROVED** [review]

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (jedupau) [reply .github/workflows/docker-publish.yml]: changed https://github.com/DataDog/test-infra-definitions/pull/849/commits/e8c34b62e04e3575f7f2e35af3ec92f7bd4f6290

**Comment by KevinFairise2** [comment]
Did you consider using the Dockerhub cache we have?
It is likely that you are not aware that it exists, but we have some documentation here if you want to give it a try: https://datadoghq.atlassian.net/wiki/spaces/ADX/pages/3740697756/E2E+-+Use+our+Dockerhub+mirror+registry+on+AWS#Use-our-private-ECR-with-pull-through-cache

**Pull request closed without merging**